tst/EnergyPlus/unit/WeatherManager.unit.cc

```diff
@@ -2214,3 +2214,384 @@ TEST_F(EnergyPlusFixture, epwHeaderTest)
+        EXPECT_EQ(0, state->dataWeatherManager->Missed.WeathCodes);
+        EXPECT_EQ(9, PresWeathConds(1));
+        EXPECT_EQ(9, PresWeathConds(2));
+        EXPECT_EQ(9, PresWeathConds(3));
+        EXPECT_EQ(9, PresWeathConds(4));
+        EXPECT_EQ(9, PresWeathConds(5));
+        EXPECT_EQ(9, PresWeathConds(6));
+        EXPECT_EQ(9, PresWeathConds(7));
+        EXPECT_EQ(9, PresWeathConds(8));
+        EXPECT_EQ(9, PresWeathConds(9));
+
+        EXPECT_EQ(1984, WYear);
+        EXPECT_EQ(1, WMonth);
+        EXPECT_EQ(5, WDay);
+        EXPECT_EQ(15, WHour);
+        EXPECT_EQ(60, WMinute);
+        EXPECT_EQ(14.4, DryBulb);
+        EXPECT_EQ(2.2, DewPoint);
+        EXPECT_EQ(44, RelHum);
+        EXPECT_EQ(103200, AtmPress);
+        EXPECT_EQ(586, ETHoriz);
+        EXPECT_EQ(1415, ETDirect);
+        EXPECT_EQ(307, IRHoriz);
+        EXPECT_EQ(357, GLBHoriz);
+        EXPECT_EQ(581, DirectRad);
+        EXPECT_EQ(115, DiffuseRad);
+        EXPECT_EQ(37300, GLBHorizIllum);
+        EXPECT_EQ(54200, DirectNrmIllum);
+        EXPECT_EQ(16400, DiffuseHorizIllum);
+        EXPECT_EQ(2230, ZenLum);
+        EXPECT_EQ(240, WindDir);
+        EXPECT_EQ(3.1, WindSpeed);
+        EXPECT_EQ(1, TotalSkyCover);
+        EXPECT_EQ(0, OpaqueSkyCover);
+        EXPECT_EQ(20.0, Visibility);
+        EXPECT_EQ(22000, CeilHeight);
+
+        EXPECT_EQ(0, PrecipWater);
+        EXPECT_EQ(0.1570, AerosolOptDepth);
+        EXPECT_EQ(0, SnowDepth);
+        EXPECT_EQ(88, DaysSinceLastSnow);
+        EXPECT_EQ(0.0, Albedo);
+        EXPECT_EQ(0.0, LiquidPrecip);
+    }
+
+    {
+        int WYear = 0;
+        int WMonth = 0;
+        int WDay = 0;
+        int WHour = 0;
+        int WMinute = 0;
+        Real64 DryBulb = 0.0;
+        Real64 DewPoint = 0.0;
+        Real64 RelHum = 0.0;
+        Real64 AtmPress = 0.0;
+        Real64 ETHoriz = 0.0;
+        Real64 ETDirect = 0.0;
+        Real64 IRHoriz = 0.0;
+        Real64 GLBHoriz = 0.0;
+        Real64 DirectRad = 0.0;
+        Real64 DiffuseRad = 0.0;
+        Real64 GLBHorizIllum = 0.0;
+        Real64 DirectNrmIllum = 0.0;
+        Real64 DiffuseHorizIllum = 0.0;
+        Real64 ZenLum = 0.0;
+        Real64 WindDir = 0.0;
+        Real64 WindSpeed = 0.0;
+        Real64 TotalSkyCover = 0.0;
+        Real64 OpaqueSkyCover = 0.0;
+        Real64 Visibility = 0.0;
+        Real64 CeilHeight = 0.0;
+        Real64 PrecipWater = 0.0;
+        Real64 AerosolOptDepth = 0.0;
+        Real64 SnowDepth = 0.0;
+        Real64 DaysSinceLastSnow = 0.0;
+        Real64 Albedo = 0.0;
+        Real64 LiquidPrecip = 0.0;
+        int PresWeathObs = 0;
+        Array1D_int PresWeathConds(9);
+        bool ErrorFound = false;
+
+        // Not providing the 5 last records
+        std::string WeatherDataLine =
+            "2018,1,5,15,0,?9?9?9?9E0?9?9?9?9?9?9?9?9?9?9?9?9?9*9?9?9?9,16.60,13.50,82,100997,588,1415,369,195,14,189,22230,1274,22230,"
+            "10584,320,10.00,9,9,777.7,1050,0,909999999,37,0.0840";
```

Comment:
An another test to make sure the refactor I need works when some of the last fields aren't provided.

BTW, I'm surprised the last number isn't processed. https://bigladdersoftware.com/epx/docs/22-2/auxiliary-programs/energyplus-weather-file-epw-data-dictionary.html

```diff
     N26, \field Present Weather Observation
     N27, \field Present Weather Codes
     N28, \field Precipitable Water
        \units mm
        \missing 999
     N29, \field Aerosol Optical Depth
        \units thousandths
        \missing .999
     N30, \field Snow Depth
        \units cm
        \missing 999
     N31, \field Days Since Last Snowfall
        \missing 99
     N32, \field Albedo
        \missing 999
     N33, \field Liquid Precipitation Depth
          \units mm
        \missing 999
-     N34; \field Liquid Precipitation Quantity
-          \units hr
-        \missing 99
```